Microsoft.Identity.Web/Client/TokenCacheProviders/InMemory/MSALPerUserMemoryTokenCacheProvider.cs

```diff
@@ -111,17 +92,19 @@ private void UserTokenCacheAfterAccessNotification(TokenCacheNotificationArgs ar
         /// <param name="args">Contains parameters used by the MSAL call accessing the cache.</param>
         private void UserTokenCacheBeforeAccessNotification(TokenCacheNotificationArgs args)
         {
-            string cacheKey = httpContextAccessor.HttpContext.User.GetMsalAccountId();
+            string cacheKey = _httpContextAccessor.HttpContext.Request.Headers["Authorization"];
```

Comment:
The issue is that the cache key is now a secret, and the cache key is not encrypted ....

Reply:
@jmprieur in the sample, in this key/value persistence, value (the serialized token cache) is not encrypted, so encrypting the key does not help because the value is not encrypted. I think this is an application specific solution if the developer wants to persist in memory or somewhere else and encrypt the cache there, they can make that decision. If we want to include encryption, it should be for both the key and value. If we want to encrypt the value, we can use the hash for the key. I think the main point here is using the access token, not the value in GetMsalAccountId() because of the reasons listed above.

---
In reply to: [318914706](https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/pull/93#discussion_r318914706) [](ancestors = 318914706)

Reply:
In fact there are some token cache serilization implementations in Microsoft.Identity.Web that do encrypt (SQL for instance).